example/tutorial/launch_bounds/launch_bounds_reduce.cpp

```diff
@@ -125,7 +126,7 @@ int main(int argc, char* argv[]) {
 
   // With LaunchBounds, we can reduce the register usage to 32
   Kokkos::parallel_reduce(
-      Kokkos::RangePolicy<Kokkos::LaunchBounds<512, 4>>(0, n), collision(),
+      Kokkos::RangePolicy<Kokkos::LaunchBounds<512, 2>>(0, n), collision(),
```

Comment:
I would signoff on the PR if it wasn't for this change.
I am not blocking but I would rather leave it as an issue to tackle in follow up work.
Did anyone check the register usage with the reduced min blocks per SM?

Reply:
```
ptxas info    : 0 bytes gmem, 32784 bytes cmem[3]
ptxas info    : Compiling entry function '_ZN6Kokkos4Impl33cuda_parallel_launch_local_memoryINS0_14ParallelReduceINS0_22CombinedFunctorReducerI9collisionNS0_15FunctorAnalysisINS0_23FunctorPatternInterface6REDUCEENS_11RangePolicyIJNS_12LaunchBoundsILj512ELj2EEEEEES4_iE7ReducerEvEESB_NS_4CudaEEELj512ELj2EEEvT_' for 'sm_86'
ptxas info    : Function properties for _ZN6Kokkos4Impl33cuda_parallel_launch_local_memoryINS0_14ParallelReduceINS0_22CombinedFunctorReducerI9collisionNS0_15FunctorAnalysisINS0_23FunctorPatternInterface6REDUCEENS_11RangePolicyIJNS_12LaunchBoundsILj512ELj2EEEEEES4_iE7ReducerEvEESB_NS_4CudaEEELj512ELj2EEEvT_
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 27 registers, 432 bytes cmem[0]
ptxas info    : Compiling entry function '_ZN6Kokkos4Impl33cuda_parallel_launch_local_memoryINS0_14ParallelReduceINS0_22CombinedFunctorReducerI9collisionNS0_15FunctorAnalysisINS0_23FunctorPatternInterface6REDUCEENS_11RangePolicyIJNS_4CudaEEEES4_iE7ReducerEvEESA_S9_EEEEvT_' for 'sm_86'
ptxas info    : Function properties for _ZN6Kokkos4Impl33cuda_parallel_launch_local_memoryINS0_14ParallelReduceINS0_22CombinedFunctorReducerI9collisionNS0_15FunctorAnalysisINS0_23FunctorPatternInterface6REDUCEENS_11RangePolicyIJNS_4CudaEEEES4_iE7ReducerEvEESA_S9_EEEEvT_
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 27 registers, 432 bytes cmem[0]
ptxas info    : Compiling entry function '_ZN6Kokkos4Impl33cuda_parallel_launch_local_memoryINS0_11ParallelForINS0_16ViewValueFunctorINS_6DeviceINS_4CudaENS_9CudaSpaceEEEjLb1EEENS_11RangePolicyIJS5_NS_9IndexTypeIlEEEEES5_EEEEvT_' for 'sm_86'
ptxas info    : Function properties for _ZN6Kokkos4Impl33cuda_parallel_launch_local_memoryINS0_11ParallelForINS0_16ViewValueFunctorINS_6DeviceINS_4CudaENS_9CudaSpaceEEEjLb1EEENS_11RangePolicyIJS5_NS_9IndexTypeIlEEEEES5_EEEEvT_
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 8 registers, 472 bytes cmem[0]
```